tripy/tripy/frontend/utils.py

```diff
@@ -28,12 +28,33 @@
 
 # Try to include correct column offsets for non-tensor arguments.
 def _add_column_info_for_non_tensor(
-    arg, arg_index, is_kwarg, dtype, num_args, func_name, skip_num_stack_entries, list_index=None
+    arg,
+    arg_index,
+    is_kwarg,
+    dtype,
+    num_args,
+    func_name,
+    skip_num_stack_entries,
+    list_index=None,
+    TensorType=None,
 ):
     from tripy.frontend.tensor import Tensor
+    from tripy.frontend.shape import Shape
+    from tripy.frontend.trace.ops.cast import cast
+
+    TensorType = utils.default(TensorType, Tensor)
+
+    assert not isinstance(
+        arg, Tensor
+    ), f"This function should not be called for objects that are already Tensor instances"
 
-    assert not isinstance(arg, Tensor)
-    arg = Tensor(arg, dtype=dtype)
+    if isinstance(arg, numbers.Number) and TensorType == Shape:
+        # Shapes require 1D values.
+        arg = [arg]
+
+    arg = TensorType(arg)
+    if dtype is not None:
+        arg = cast(arg, dtype=dtype)
```

Comment:
Perhaps we should disallow this branch or have an assert related to dtype for Shape/ShapeScalar since those have to be int32. Not strictly necessary.

Reply:
I think we can actually allow other types since `cast` will then return a `Tensor` instead, which may still work with the operation.

Reply:
True. I think that's fine and wouldn't lead to surprising behavior, since the only casting is with `sync_dtype`.